### DISCUSSION TITLE
extract lenght

### DISCUSSION THREAD

**Discussion by GaiaBes** [Abe404/root_painter#18] [Q&A] [answered]
HI! It is me again! I have one more question.
I am analyzing pictures of roots in soils profiles. 
When I extract lenght from my segmented pictures it gives me back an Excel sheet with the name of the photo followed by a number, which i think it is in pixels and need to be transformed in centimeters.
1cm is  37.795275591 pixel with a risolution of 96dpi so for example i have 3087 as the extracted root lenght from the first picture which means: 3087/37.795 is 81,688 cm. The total root lenght in the first picture is 81,69 cm. Is it the right procedure?

and so the number given is the total lenght of all the roots in the pictures right? can i use it to calculate root lenght density? 
And if i divided it by the number of roots counted in each picture with the option extract count I should get the average lenght of roots in each picture? 

Sorry for the many questions
Kind regards

**Comment by Abe404** [comment, marked as answer]
HI Gaia,

> When I extract lenght from my segmented pictures it gives me back an Excel sheet with the name of the photo followed by a number, which i think it is in pixels and need to be transformed in centimeters.

Yes it is in pixels as RootPainter doesn't have information on the physical dimensions of the images.

> 1cm is 37.795275591 pixel with a risolution of 96dpi so for example i have 3087 as the extracted root lenght from the first picture which means: 3087/37.795 is 81,688 cm. The total root lenght in the first picture is 81,69 cm. Is it the right procedure?

Approximately :)

If 1cm = 37.795275591 pixels, then 1 pixel = 1/37.795275591 = 0.02645833333 cm. 
If the expected root length is 3087 then that means it is (3097 * 0.02645833333) cm = 81.941458323 cm

> and so the number given is the total lenght of all the roots in the pictures right? can i use it to calculate root lenght density?

Yes it's for the total length of roots per image. I haven't calculated RLD myself but it should be possible.  Perhaps you could just divide by the image area to get RLD for the image? 

> And if i divided it by the number of roots counted in each picture with the option extract count I should get the average lenght of roots in each picture?

Yes.

In the new version of RootPainter there is an option from the extras menu to convert the RootPainter output segmentations into a format that is convenient to use for analysis in RhizoVision Explorer (RVE). I suggest giving RVE a try as it will open up more possibilities in terms of the root traits you can work with:
https://www.biorxiv.org/content/10.1101/2021.04.11.439359v1.full

In addition, depending on your research question, some of the methods used by Alonso-Crespo et al for analysing RootPainter segmentations may also be of interest:
https://www.biorxiv.org/content/10.1101/2021.08.24.457510v1.full

Thanks for the questions - Hopefully this information can be useful for others too :) 

Kind regards,
Abraham
